packages/eas-cli/src/commandUtils/builds.ts

```diff
@@ -0,0 +1,68 @@
+import chalk from 'chalk';
+
+import { Build, BuildFragment, BuildStatus } from '../graphql/generated';
+import { BuildQuery } from '../graphql/queries/BuildQuery';
+import { appPlatformEmojis } from '../platform';
+import { ExpoGraphqlClient } from './context/contextUtils/createGraphqlClient';
+
+export async function ensureBuildExistsAsync(
+  graphqlClient: ExpoGraphqlClient,
+  buildId: string
+): Promise<void> {
+  try {
+    await BuildQuery.byIdAsync(graphqlClient, buildId);
+  } catch {
+    throw new Error(`Couldn't find a build matching the id ${buildId}`);
+  }
+}
+
+export async function fetchBuildsAsync(
+  graphqlClient: ExpoGraphqlClient,
+  projectId: string,
+  statuses?: BuildStatus[]
+): Promise<BuildFragment[]> {
+  let builds = [];
+  if (!statuses) {
+    builds = await BuildQuery.viewBuildsOnAppAsync(graphqlClient, {
+      appId: projectId,
+      offset: 0,
+      limit: 10,
+    });
+  } else {
+    for (const status of statuses) {
+      const buildsForStatus = await BuildQuery.viewBuildsOnAppAsync(graphqlClient, {
+        appId: projectId,
+        offset: 0,
+        limit: 10,
+        filter: { status },
+      });
+      builds.push(...buildsForStatus);
```

Comment:
Should we order them by `createdAt`?

Reply:
I think they are sorted like that implicitly, but need to double check

Reply:
They were, but it fell apart when combining results for different statuses, as this would group the builds by status too. Added explicit sorting before returning